test/ProductTest.java

```diff
@@ -0,0 +1,25 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import products.Brand;
+import products.Product;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ProductTest {
+
+    @Test
+    public void takeTheProductsTest() throws SQLException {
+        Product product= new Product(Brand.APPLE, "4x4",135.00, "è un pc molto veloce, potente", "computer", 50);
+
+        List<Product> list= new ArrayList<>();
```

Comment:
non capisco come testi questo metodo, al massimo puoi crearti un oggetto uguale a quello del db e fare la verifica tramite equals, se si equivalgono il test funziona,sistema questa cosa

Reply:
ho creato un db in locale e ho provato ad aggiungere nel MIO db alcuni prodotti, e tramite il test, funziona.

Reply:
prima di richiamare il tuo metodo, richiama addProductOnDB(). il test ti passa giusto perché hai inserito tu dei prodotti, ma adesso che hai il metodo di giuseppe il test va fatto inserendoli tramite addProductOnDB().